### ISSUE TITLE
Enhancement: [no-floating-promises] Disable checkThenables by default in v8

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#9508]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://v8--typescript-eslint.netlify.app/rules/no-floating-promises

### Description

Following #8433 -> #9263: we'd resolved to add a new `checkThenables` option to the `no-floating-promises` rule. It's enabled by default in v7 to avoid a breaking change. But for v8 we'd like to disable it by default.

### Fail

```TypeScript
/* @typescript-eslint/no-floating-promises: ["error", { "checkThenables": true }] */

interface MyThenable extends PromiseLike<string> {}

declare function createMyThenable(): Promise<string>;

createMyThenable();
```

### Pass

```TypeScript
interface MyThenable extends PromiseLike<string> {}

declare function createMyThenable(): MyThenable;

createMyThenable();
```

### Additional Info

💖
